src/contract/nodes.rs

```diff
@@ -60,23 +63,41 @@ impl sha256t::Tag for NodeIdTag {
         sha256::HashEngine::from_midstate(midstate, 64)
     }
 }
-
-// TODO #48: Refactor all ids into a single style after `ConsignmentId`
-/// Unique node (genesis, extensions & state transition) identifier equivalent
-/// to the commitment hash
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
 #[derive(
-    Wrapper, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default, From,
-)]
-#[wrapper(
-    Debug, Display, LowerHex, Index, IndexRange, IndexFrom, IndexTo, IndexFull
+    Wrapper,
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Default,
+    From,
+    Display,
 )]
+#[wrapper(Debug, LowerHex, Index, IndexRange, IndexFrom, IndexTo, IndexFull)]
+#[display(NodeId::to_bech32_string)]
 pub struct NodeId(sha256t::Hash<NodeIdTag>);
```

Comment:
This will not work with serde serialization and will produce hex-encoded string. A special serde internal attribute macro must be used in order to do correct serialization for YAML/JSON serde encodings